src/sedpack/io/file_info.py

```diff
@@ -52,3 +54,143 @@ def no_directory_traversal(cls, v: Path) -> Path:
+    ca27/f469/ca5a8cf3aad942828dd9f8e56f2fbd42
+    ca27/f469/ad236737fa9041318e13ac16a6400188
+    9c40/7a10/e0151d9241a049d883bd985ab256c8d2
+    9c40/7a10/3e1712e509c04630b7a4f690b83d2f92
+    9c40/c1d9/75fff313cd1b479dbe3beb5eb78d8598
+    9c40/c1d9/aa4a91ffd1514e2d97cbeafb7cac25e0
+    2f66/3b68/c1139fb2669e456a85310ae9170dc66e
+    2f66/3b68/ee6dbbf7d4b24a23a2126d5bc607da16
+    ```
+    which results in the following directory structure:
+    ```bash
+    ├── 2f66
+    │   └── 3b68
+    │       ├── c1139fb2669e456a85310ae9170dc66e
+    │       └── ee6dbbf7d4b24a23a2126d5bc607da16
+    ├── 9c40
+    │   ├── 7a10
+    │   │   ├── 3e1712e509c04630b7a4f690b83d2f92
+    │   │   └── e0151d9241a049d883bd985ab256c8d2
+    │   └── c1d9
+    │       ├── 75fff313cd1b479dbe3beb5eb78d8598
+    │       └── aa4a91ffd1514e2d97cbeafb7cac25e0
+    └── ca27
+        ├── ac58
+        │   ├── ab29ddcb515f4cf1b3651cab31490142
+        │   └── f37c107d9a3549b68fba18e031aa1cf9
+        └── f469
+            ├── ad236737fa9041318e13ac16a6400188
+            └── ca5a8cf3aad942828dd9f8e56f2fbd42
+    ```
+    """
+
+    def __init__(self,
+                 levels: int = 4,
+                 max_branching: int = 1_000,
+                 name_length: int = 10) -> None:
+        """Initialize a top level generator. No collision detection since in
+        theory there could be other files present. The last part of the path is
+        interpreted as the file-name without a file-extension.
+
+        Args:
+
+          levels (int): How many levels of directory hierarchy to generate.
+          Defaults to 4.
+
+          max_branching (int): How many subdirectories to generate (top level
+          can generate unlimited). Defaults to 1_000.
+
+          name_length (int): Maximum length of one path part. The last part is
+          length 32 always to avoid collisions in the file-name. Make sure that
+          `max_branching` is not too large when selecting small `name_length`.
+          The name consists of hex digits so `16**name_length` should be much
+          larger than `math.sqrt(max_branching)` otherwise a collision is
+          probable. See `https://en.wikipedia.org/wiki/Birthday_problem`.
+        """
+        assert levels >= 1
+        assert max_branching > 1
+        assert name_length > 1
+
+        self.levels: int = levels
+        self.max_branching: int = max_branching
+        self.name_length: int = name_length
+
+        # The top level object is allowed to have arbitrary degree and never
+        # raises.
+        self.is_top: bool = True
+        # How many part names has this object generated. Updated by `_get_name`.
+        self.generated: int = 0
+        # The current part name. Regenerated by `_get_name`.
+        self.current_directory: str
+
+        self.next_level: DirectoryGenerator | None = None
+        if levels > 1:
+            self._refresh()
+
+    def get_path(self) -> str:
```

Comment:
For portability, I think it would be safer to return a Path object instead of a string. They are immutable, like strings but will handle the OS differences much better and you can turn them into string only for json serialization.

Reply:
done